### PR TITLE
refactor css variables

### PR DESCRIPTION
**Pull request by Kalmaegi**
For https://github.com/microsoft/vscode/issues/165169
- [x] src/vs/workbench/browser/parts/activitybar/activitybarActions.ts
- [x]  src/vs/workbench/contrib/testing/browser/theme.ts

### CODE REVIEW

**Comment by weartist** [comment]
@aeschli need your help, thanks

**Comment by weartist** [comment]
@aeschli hi，How can I request a review?

**Review by aeschli — DISMISSED** [review]

**Review by jrieken — DISMISSED** [review]

**Comment by weartist** [comment]
@aeschli  need your help, thanks~

File: src/vs/workbench/contrib/testing/browser/theme.ts [hunk 1]
```diff
@@ -111,3 +118,14 @@ export const testStatesToIconColors: { [K in TestResultState]?: string } = {
 	[TestResultState.Unset]: testingColorIconUnset,
 	[TestResultState.Skipped]: testingColorIconSkipped,
 };
+
```
Comment (aeschli) [inline src/vs/workbench/contrib/testing/browser/theme.ts:121]: Why is this added?

File: src/vs/workbench/browser/parts/activitybar/media/activityaction.css [hunk 2]
```diff
@@ -200,3 +200,46 @@
...
+.monaco-workbench .activitybar > .content :not(.monaco-menu) > .monaco-action-bar .action-item:hover .action-label.profile-activity-item {
+	color: var(--vscode-activityBarItem-profilesHoverForeground) !important;
+}
+
+.monaco-workbench .activitybar > .content :not(.monaco-menu) > .monaco-action-bar .action-item.checked .active-item-indicator:before {
+	border-left-color: var(--vscode-activityBar-activeBorder);
+}
+
+.monaco-workbench .activitybar > .content :not(.monaco-menu) > .monaco-action-bar .action-item.checked:focus::before {
+	visibility: hidden;
```
Comment (aeschli) [inline src/vs/workbench/browser/parts/activitybar/media/activityaction.css:233]: This rule must only be active if ACTIVITY_BAR_ACTIVE_FOCUS_BORDER is not null....

File: src/vs/workbench/browser/parts/activitybar/media/activityaction.css [hunk 3]
```diff
@@ -200,3 +200,46 @@
...
+}
+
+.monaco-workbench .activitybar > .content :not(.monaco-menu) > .monaco-action-bar .action-item.checked:focus .active-item-indicator:before {
+	visibility: visible;
+	border-left-color: var(--vscode-activityBar-activeFocusBorder);
+}
+
+.monaco-workbench .activitybar > .content :not(.monaco-menu) > .monaco-action-bar .action-item.checked .active-item-indicator {
+	z-index: 0;
```
Comment (aeschli) [inline src/vs/workbench/browser/parts/activitybar/media/activityaction.css:242]: Not sure if it matters, but `z-index: 0;` is now always set, Before it was only set when ACTIVITY_BAR_ACTIVE_BACKGROUND was not null

**Review by aeschli — CHANGES_REQUESTED** [review]

**Review by bpasero — CHANGES_REQUESTED** [review]
I think Martin already reviewed the problematic parts: some of the properties are conditional based on the color being present or not, so we cannot just move them into CSS. In those cases we probably need to apply a dedicated class to the element to toggle these styles that cannot be expressed as variables.

Besides, not all of the rules have been extracted from activity bar actions it seems.

File: src/vs/workbench/browser/parts/activitybar/media/activityaction.css (hunk 2 above)
Comment (weartist) [reply src/vs/workbench/browser/parts/activitybar/media/activityaction.css:233]: yes, It was my negligence

**Comment by weartist** [comment]
@aeschli @bpasero 
Sorry, something went wrong with conflict resolution halfway through, my oversight, for conditional, can we use variables to determine, e.g:

image

If activitybarActions.ts doesn't need to be modified for now, we'll close this pr, I'm really sorry for this blunder.
